### PR TITLE
Change the order of indices in Batchnorm

### PR DESCRIPTION
**Pull request by mariogeiger**
- changed the order of indices `[batch, features, spacial]` -> `[batch, spacial, features]`
- complete docstring

### CODE REVIEW

File: tests/batchnorm_test.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class Tests(unittest.TestCase):
+    def test_that_it_runs(self):
+        Rs = [(3, 1), (4, 3)]
+        m = BatchNorm(Rs)
+
+        m.to(dtype=torch.float64)
+        x = torch.randn(10, sum(mul * d for mul, d in Rs), dtype=torch.float64)
+        m(x)
```
Comment (blondegeek) [inline tests/batchnorm_test.py]: Should we be testing any properties of the output?

File: tests/batchnorm_test.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+        m.to(dtype=torch.float32)
```
Comment (blondegeek) [inline tests/batchnorm_test.py]: Rather than just two examples, do we want to add 10 for test?
Additionally, should we test that on average, for inputs of varying variance (perhaps an additional outside look to the number of tests), that batch norm is able to bring the variance back to 1?

File: tests/batchnorm_test.py (hunk 1 above)
Comment (mariogeiger) [reply tests/batchnorm_test.py]: Yes we should do more than that :+1:

File: tests/batchnorm_test.py (hunk 2 above)
Comment (mariogeiger) [reply tests/batchnorm_test.py]: You are right, it was laziness

File: tests/batchnorm_test.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        m = BatchNorm(Rs, normalization='component')
+
+        x = torch.randn(batch, n, sum(mul * d for mul, d in Rs)).mul(5.0).add(10.0)
+        x = m(x)
+
+        a = x[..., :3]  # [batch, space, mul]
+        assert a.mean([0, 1]).abs().max() < 1e-10
+        assert a.pow(2).mean([0, 1]).sub(1).abs().max() < 1e-5
+
+        a = x[..., 3:].reshape(batch, n, 4, 3)  # [batch, space, mul, repr]
```
Comment (blondegeek) [inline tests/batchnorm_test.py]: Super cool. Thanks for the tests. Should there also be checks on variance or is mean sufficient?

File: e3nn/batchnorm.py [hunk 4]
```diff
@@ -28,8 +39,14 @@ def __init__(self, Rs, eps=1e-5, momentum=0.1, affine=True, reduce='mean'):
             self.register_parameter('weight', None)
             self.register_parameter('bias', None)
 
+        assert isinstance(reduce, str), "reduce should be passed as a string value"
+        assert reduce in ['mean', 'max'], "reduce needs to be 'mean' or 'max'"
         self.reduce = reduce
 
+        assert isinstance(normalization, str), "normalization should be passed as a string value"
+        assert normalization in ['norm', 'component'], "normalization needs to be 'norm' or 'component'"
```
Comment (blondegeek) [inline e3nn/batchnorm.py]: Ah -- nice feature!
